ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj

```diff
@@ -361,6 +376,28 @@
   (modify-checksum*
    context concept bulk-update-params user-id update-checksum))
 
+(defn- modify-size*
+  "Add or update the size"
+  [context concept bulk-update-params user-id xf]
+  (let [{:keys [format metadata]} concept
+        {:keys [granule-ur new-value]} bulk-update-params
+        ;; invoke the appropriate transform - for size, there is only one option (update)
+        updated-concept (xf context concept new-value)
+        {updated-metadata :metadata updated-format :format} updated-concept]
+    (if-let [err-messages (:errors updated-metadata)]
+      (errors/throw-service-errors :invalid-data err-messages)
+      (-> concept
+          (assoc :metadata updated-metadata)
+          (assoc :format updated-format)
+          (update :revision-id inc)
+          (assoc :revision-date (time-keeper/now))
+          (assoc :user-id user-id)))))
```

Comment:
This can probably be coalesced with `modify-checksum` and the to-be-implemented `modify-format`